### PR TITLE
Release 2.4.0

### PR DESCRIPTION
**Pull request by BSchilperoort**
- Followed all steps outlined in CONTRIBUTING.md https://github.com/eWaterCycle/ewatercycle/blob/main/CONTRIBUTING.md#you-want-to-make-a-release
- Build failed with new conda lock file. Its resolver picked an old pyopenssl version incompatible with the openssl binaries it picked :roll_eyes: . Fixed by pinning newer version in dependencies

### CODE REVIEW

**Comment by BSchilperoort** [comment]
All packages need to be on conda to make a new conda lock file. ~~Remote BMI is not on conda yet.~~ Done!

File: pyproject.toml [hunk 1]
```diff
@@ -52,7 +52,8 @@ dependencies = [
     "Shapely",
     "xarray",
     "fsspec",
-    'cartopy',
+    "cartopy",
+    "pyopenssl>=24.0.0",  # otherwise conda lock file will have a too old version
```
Comment (Peter9192) [inline pyproject.toml]: Perhaps link to https://github.com/conda/conda/issues/13619?

**Review by Peter9192 — APPROVED** [review]
Looks allright to me. Added a small suggestion.

File: pyproject.toml (hunk 1 above)
Comment (BSchilperoort) [reply pyproject.toml]: Yeah, doesn't hurt to add that.

**Comment by BSchilperoort** [comment]
Thanks for the review @Peter9192 !

**Pull request merged**